docker/docker.go

```diff
@@ -88,3 +100,11 @@ func isFile(path string) bool {
 
 	return fileInfo.Mode().IsRegular()
 }
+
+func validateFlags(flags *BuildFlags) error {
+	engines := []string{"", "docker", "podman"}
```

Comment:
Would we want to let the user specify whatever binary they want?

- Pros:
	- We could handle scenarios where the docker or podman binary is named something else (does this happen with multi-version installs?)
	- It doesn't need to be in the user's PATH.
- Cons:
	- It's easier to make the flag more restrictive at first and loosen it later if we need to.
	- We don't have control over what the build arguments should be, so all the engines would need to support the same exact flags.
	- Easier for the user to make typo's/mistakes that might lead to cryptic errors.

Reply:
Yeah, my thought was to start restricted, and if it works well and people need custom binaries, we let it be arbitrary.